### ISSUE TITLE
System.Text.Json deserialization bug

### ISSUE THREAD

**Issue by ngiakhanh** [dotnet/runtime#61550]
### Describe the bug
I serialize an instance, and then deserialize it again. I receive deserialized instance which is different than the initial instance

### To Reproduce

See it in fiddle here https://dotnetfiddle.net/TiwIrj

### Further technical details
- ASP.NET Core version: 6

**Comment by willdean** [comment]
1. In `BlockChain`, you need to remove the `private` in the setter of `public List<Transaction> PendingTransactions { get; private set; }`, because you're not giving the deserializer a way to set this property.

2.  You need to do the same thing for `public List<Block> Chain { get; private set; }` for the same reason

3. In the constructor of `Block` you need to change `IEnumerable<Transaction> transactions` to `IList<Transaction> transactions` so that it matches the type of the `Transactions` property.

Basically the deserializer doesn't use any kind of magic to set things - it either needs *public* properties or it needs constructor arguments which closely match the property types they're supposed to be deserializing.

**Comment by ngiakhanh** [comment]
Thanks for your explanation!
